lib/ansible/modules/cloud/amazon/elasticache_replication_group.py

```diff
@@ -0,0 +1,585 @@
+
+    new_node_id = max([int(cc['CacheNodeId']) for cc in rep_group['NodeGroups'][0]['NodeGroupMembers']]) + 1
+    while module.params['num_cache_clusters'] > current_cache_cluster_count:
+        try:
+            # Cache Cluster Ids must be less than 20 letters and not contain two consecutive hyphens
+            # The replace is needed when the 16th character of the replication group happens to be a hyphen
+            new_cache_cluster_id = ("%s-%03d" % (rep_group['ReplicationGroupId'][:16], new_node_id)).replace('--', '-')
+            connection.create_cache_cluster(CacheClusterId=new_cache_cluster_id,
+                                            ReplicationGroupId=rep_group['ReplicationGroupId'])
+            new_node_id += 1
+            current_cache_cluster_count += 1
+            changed = True
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, "Couldn't add new cache cluster %s" % new_cache_cluster_id)
+        if module.params.get('wait'):
+            waiter = connection.get_waiter('cache_cluster_available')
+            waiter.wait(CacheClusterId=new_cache_cluster_id)
+
+    # FIXME: Handle both security group Ids and Names for SGs
+
+    if params:
+        params['ReplicationGroupId'] = module.params.get('name')
+        params['ApplyImmediately'] = module.params.get('apply_immediately')
+
+        try:
+            connection.modify_replication_group(**params)
+            changed = True
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            msg = "Couldn't modify replication group"
+            if e.response['Error']['Code'] == 'InvalidParameterCombination':
+                msg = msg + (". This might be because of unsupported settings for T1/T2 instances. See: "
+                             "http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/ParameterGroups.Redis.html#ParameterGroups.Redis.NodeSpecific"
+                             " for more details")
+            module.fail_json_aws(e, msg=msg)
+
+    result = check_for_rep_group(module, connection)
+    module.exit_json(changed=changed, warnings=warnings, **camel_dict_to_snake_dict(result))
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            auto_minor_version_upgrade=dict(default=False, type='bool'),
+            name=dict(required=True),
+            description=dict(),
+            primary_cluster_id=dict(),
+            failover_enabled=dict(default=False, type='bool'),
+            apply_immediately=dict(default=False, type='bool'),
+            num_cache_clusters=dict(type='int'),
+            preferred_cache_cluster_azs=dict(type='list'),
+            node_type=dict(),
+            engine=dict(default='redis', choices=['redis']),
+            cache_engine_version=dict(),
+            cache_parameter_group=dict(),
+            cache_subnet_group_name=dict(),
```

Comment:
As I mentioned above, this should be `cache_subnet_group_names=dict(type='list'),`